assertions/src/test/java/io/camunda/zeebe/process/test/assertions/BpmnAssertTest.java

```diff
@@ -0,0 +1,100 @@
+package io.camunda.zeebe.process.test.assertions;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.mock;
+
+import io.camunda.zeebe.client.api.response.ActivatedJob;
+import io.camunda.zeebe.client.api.response.DeploymentEvent;
+import io.camunda.zeebe.client.api.response.ProcessInstanceEvent;
+import io.camunda.zeebe.client.api.response.ProcessInstanceResult;
+import io.camunda.zeebe.client.api.response.PublishMessageResponse;
+import io.camunda.zeebe.process.test.api.RecordStreamSource;
+import io.camunda.zeebe.process.test.inspections.model.InspectedProcessInstance;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class BpmnAssertTest {
+
+  @BeforeEach
+  public void beforeEach() {
+    BpmnAssert.initRecordStream(mock(RecordStreamSource.class));
+  }
+
+  @AfterEach
+  public void afterEach() {
+    BpmnAssert.resetRecordStream();
+  }
+
+  @Test
+  public void testAssertThatProcessInstanceEventReturnsProcessInstanceAssert() {
+    // given
+    ProcessInstanceEvent event = mock(ProcessInstanceEvent.class);
+
+    // when
+    ProcessInstanceAssert assertions = BpmnAssert.assertThat(event);
+
+    //then
+    assertThat(assertions).isInstanceOf(ProcessInstanceAssert.class);
```

Comment:
The test would not compile if `BpmnAssert.assertThat()` would return something else than `ProcessInstanceAssert`.

:wrench: We could modify the test to fail in the assertion instead of not compiling.

```suggestion
    final var assertions = BpmnAssert.assertThat(event);

    //then
    assertThat(assertions).isInstanceOf(ProcessInstanceAssert.class);
```

Reply:
Unfortunately we can't do this as this module should support Java 8